File: Python Basic/Circle.py
import math 

class Circle:
    # Construct a circle object 
    def __init__(self, radius = 1):
        self.radius = radius

    def getPerimeter(self):
        return 2 * self.radius * math.pi

    def getArea(self):
        return self.radius * self.radius * math.pi
          
    def setRadius(self, radius):
        self.radius = radius

File: JAVA/chapter9_examples/chapter9_examples/Circle.java
package chapter9_examples;

public class Circle {
	private double radius; // The radius
	private static int numberOfObjects = 0; // Number of objects created 
	
	// Construct a circle with radius 1
	public Circle() {
		this(1.0);
	}

	// Construct a circle with the specified radius
	public Circle(double radius) {
		this.radius = radius;
		numberOfObjects++;
	}
	
	// Return the radius
	public double getRadius() {
		return radius;
	}
	
	// Set a new radius
	public void setRadius(double radius) {
		this.radius = radius;
	}
	
	// Return the number of objects created
	public static int getNumberOfObjects() {
		return numberOfObjects;
	}

	// Return the area
	public double getArea() {
		return radius * radius * Math.PI;
	}
	
	// Return a string representation
	public String toString() {
		return "radius = " + radius;
	}
}
